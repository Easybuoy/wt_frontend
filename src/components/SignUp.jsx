import React from "react";
import styled from "styled-components";
import { Button } from "@chakra-ui/core";
import banner from "./assets/banner.jpg";

const SignUpStyle = styled.div`

    width: 100%;
    height: 100vh;
    color: black;
    background-color:white;



    .signup-container{
        width: 100%;
        height: 100%;
        display: flex;
    }

   .signup-banner{
       width: 50%;
       opacity: 0.8;

        @media only screen and (max-width: 800px){
            display: none;
        }

       img{
           width: 100%;
           height:100%;
       }
    }

    .signup-form{
        width: 50%;
        display: flex;
        align-items: center;
        padding: 0 4rem;
        

        @media only screen and (max-width: 900px){
            width: 100%;
            text-align: centre;
            border: 1px solid red;
        }

        form{
            height: auto;
            width: 100%;
            text-align: left;
            padding: 2rem;
            display: flex;
            flex-direction: column;
            justify-content: center;
            

            @media only screen and (max-width: 900px){
                height: 100%;
                border: 1px solid red;
                align-items: center;
            }
            

            a{
                font-size: 16px;
                margin-top: 1rem;
                color: orange;
            }

            a:hover{
                color: grey;
            }

            input{
                display: block
                margin-top: 1rem;
                width: 80%;
                background-color: cream;

                @media only screen and (max-width: 800px){
                    margin: 2rem 0;
                }

            }

            input[type=text], input[type=email], input[type=password]{
                background-color: #fffdd0;
            }

            h1{
                font-family: ubuntu;
                font-weight: bold;
                margin-bottom:2rem;
            }

           

            .signup-form-button{
                width: 80%;
                margin: 1rem 0;
            }


            .signup-linked-button{
                width: 45%;
                margin: 1rem 1rem 0 0;
            }
            
        .signup-linked-profiles{
            width: 80%;
        }

    }


`;

function SignUp() {
  return (
    <SignUpStyle>
      <div className="signup-container">
        <div className="signup-banner">
          <img src={banner} alt="banner" />
        </div>
        <div className="signup-form">
          <form>
            <h1>Sign up</h1>
            <input placeholder="FIRST NAME" name="FIRST NAME" type="text" />
            <input placeholder="LAST NAME" name="LAST NAME" type="text" />
            <input placeholder="EMAIL" name="EMAIL" type="email" />
            <input placeholder="PASSWORD" name="PASSWORD" type="password" />
            <Button
              className="signup-form-button"
              variantColor="orange"
              rightIcon="arrow-forward"
            >
              Sign up
            </Button>
            <div className="signup-linked-profiles">
              <Button
                className="signup-linked-button"
                variantColor="orange"
                rightIcon="arrow-forward"
              >
                Google
              </Button>
              <Button
                className="signup-linked-button"
                variantColor="orange"
                rightIcon="arrow-forward"
              >
                Facebook
              </Button>
            </div>
            <a href="#">Already have an account?</a>
          </form>
        </div>
      </div>
    </SignUpStyle>
  );
}

export default SignUp;
