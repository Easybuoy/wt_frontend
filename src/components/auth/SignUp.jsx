import React, { useRef } from "react";
import { Link, Redirect } from "react-router-dom";
import { withApollo } from "react-apollo";
import GoogleLogin from "react-google-login";
import { Button, useToast } from "@chakra-ui/core";
import banner from "../../assets/banner.png";
import SignUpStyle from "../../styles/SignupStyles";
import { GOOGLE_AUTH_MUTATION, SIGNUP_MUTATION } from "../../graphql/mutations";
const { REACT_APP_GOOGLE_CLIENT_ID } = process.env;

function SignUp({ client, history }) {
  const toast = useToast();
  const firstname = useRef();
  const lastname = useRef();
  const password = useRef();
  const email = useRef();

  function onSubmit(e) {
    e.preventDefault();

    client
      .mutate({
        mutation: SIGNUP_MUTATION,
        variables: {
          firstname: firstname.current.value,
          lastname: lastname.current.value,
          password: password.current.value,
          rePassword: password.current.value,
          email: email.current.value
        }
      })
      .then(() => {
        toast({
          title: "Sign up Successful.",
          description: "Login with account details.",
          status: "success",
          duration: 9000,
          isClosable: true
        });
        history.push("/login");
      })
      .catch(error => {
        toast({
          title: "Error Sigin you up",
          description: error.graphQLErrors[0].message,
          status: "error",
          duration: 9000,
          isClosable: true
        });
      });
  }

  const responseFailureGoogle = error => {
    toast({
      title: "An error occurred.",
      description: "Unable to login to your account.",
      status: "error",
      duration: 9000,
      isClosable: true
    });
  };

  const responseGoogle = response => {
    client
      .mutate({
        mutation: GOOGLE_AUTH_MUTATION,
        variables: {
          accessToken: response.accessToken
        }
      })
      .then(res => {
        const { token, isNewUser } = res.data.authGoogle;
        localStorage.setItem("userData", JSON.stringify({ token, isNewUser }));
        if (isNewUser === true) {
          history.push("/onboarding");
        } else {
          history.push("/");
        }

        toast({
          title: "Sign in Successful.",
          description: "We've created your account for you.",
          status: "success",
          duration: 9000,
          isClosable: true
        });
      })
      .catch(error => {
        toast({
          title: "An error occurred.",
          description: "Unable to login to your account.",
          status: "error",
          duration: 9000,
          isClosable: true
        });
      });
  };

  return (
    <SignUpStyle>
      <div className="signup-container">
        <div className="signup-banner">
          <img src={banner} alt="banner" />
        </div>
        <div className="signup-form">
          <form>
            <h2>Sign up</h2>
            <input
              ref={firstname}
              required
              placeholder="FIRST NAME"
              name="FIRST NAME"
              type="text"
            />
            <input
              ref={lastname}
              required
              placeholder="LAST NAME"
              name="LAST NAME"
              type="text"
            />
            <input
              ref={email}
              required
              placeholder="EMAIL"
              name="EMAIL"
              type="email"
            />
            <input
              ref={password}
              required
              placeholder="PASSWORD"
              name="PASSWORD"
              type="password"
            />
            <input
              ref={password}
              required
              placeholder="RE-ENTER PASSWORD"
              name=" PASSWORD"
              type="password"
            />

            <Button
              type="submit"
              className="signup-form-button"
              variantColor="orange"
              rightIcon="arrow-forward"
              onClick={onSubmit}
              size="lg"
            >
              Sign up
            </Button>
            <div className="signup-linked-profiles">
              <GoogleLogin
                clientId={REACT_APP_GOOGLE_CLIENT_ID}
                render={renderProps => (
                  <Button
                    onClick={renderProps.onClick}
                    disabled={renderProps.disabled}
                    color="white"
                    bg="#4C8BF5"
                    rightIcon="arrow-forward"
                    width="45%"
                    size="lg"
                  >
                    Sign Up with Google
                  </Button>
                )}
                buttonText="Login"
                onSuccess={responseGoogle}
                onFailure={responseFailureGoogle}
                cookiePolicy={"single_host_origin"}
              />

              <Button
                type="submit"
                variantColor="facebook"
                rightIcon="arrow-forward"
                size="lg"
                width="45%"
              >
                Facebook
              </Button>
            </div>
            <Link to="/login">Already have an account?</Link>
          </form>
        </div>
      </div>
    </SignUpStyle>
  );
}

export default withApollo(SignUp);
