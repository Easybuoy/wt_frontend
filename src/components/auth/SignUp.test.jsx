import React from "react";
import { shallow } from "enzyme";

import SignUp from "./SignUp";

describe("<SignUp />", () => {
  let wrapper;

  const props = {
    location: { search: jest.mock() },
    client: { mutate: jest.fn(), query: jest.fn() },
    history: { push: jest.fn() }
  };

  beforeEach(() => {
    wrapper = shallow(<SignUp {...props} />);
  });

  it("renders the SignUp component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
