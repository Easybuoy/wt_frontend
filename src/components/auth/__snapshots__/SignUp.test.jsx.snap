// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> renders the SignUp component correctly 1`] = `
<ApolloConsumer>
  <Component />
</ApolloConsumer>
`;
